pkg/target/manager.go

```diff
@@ -282,8 +292,122 @@ func (m *managerImpl) validateSeed(ctx context.Context, seed *gardencorev1beta1.
+	for garden, gDomain := range domains {
+		if strings.Contains(domain, gDomain) {
+			return garden, domain
+		}
+	}
+
+	return "", ""
+}
+
+func targetFlagsFromDomain(ctx context.Context, m *managerImpl, t *targetImpl, name string) (TargetFlags, error) {
+	dashboardDomains, shootDomains, err := domainsFromConfiguration(m)
+	if err != nil {
+		return nil, fmt.Errorf("invalid domain format in configured gardens: %w", err)
+	}
+
+	if len(dashboardDomains) == 0 && len(shootDomains) == 0 {
+		return nil, fmt.Errorf("no domain found in configured gardens. Shoot targeting via domain requires shoot and / or dashboard domain configuration for each garden that you want to enable domain targeting for")
+	}
+
+	garden, domain := matchGardenDomain(dashboardDomains, name)
+	if domain != "" {
+		domainRegexp, err := regexp.Compile(`.+/namespace/([^/]+)/shoots/([^/]+)/?`)
+		if err != nil {
+			return nil, fmt.Errorf("could not extract project and shoot from url: %w", err)
+		}
+
+		match := domainRegexp.FindStringSubmatch(domain)
+		if match != nil {
+			gardenClient, err := m.clientForGarden(t.Garden)
+			if err != nil {
+				return nil, fmt.Errorf("could not create Kubernetes client for garden cluster: %w", err)
+			}
+
+			namespace, err := m.resolveProjectNamespace(ctx, gardenClient, match[1])
+			if err != nil {
+				return nil, fmt.Errorf("failed to validate project: %w", err)
+			}
+
+			if namespace == nil {
+				return nil, fmt.Errorf("invalid namespace in shoot url")
+			}
+
+			projectName := namespace.Labels["project.gardener.cloud/name"]
+			if projectName == "" {
+				return nil, fmt.Errorf("namespace in url is not related to a gardener project")
+			}
+
+			return NewTargetFlags(garden, projectName, "", match[2]), nil
+		}
+	}
+
+	garden, domain = matchGardenDomain(shootDomains, name)
+	if domain != "" {
+		domainRegexp, err := regexp.Compile(`([^.]+).([^.]+)\..+`)
+		if err != nil {
+			return nil, fmt.Errorf("could not extract project and shoot from url: %w", err)
+		}
+
+		match := domainRegexp.FindStringSubmatch(domain)
+		if match != nil {
+			return NewTargetFlags(garden, match[2], "", match[1]), nil
+		}
+	}
+
+	return nil, nil
+}
+
+func isDomainFormat(name string) bool {
+	// as shoots must not contain dots we can assume that the user wants to target via domain
+	match, err := regexp.MatchString(`.+\..+`, name)
+	return err == nil && match
+}
+
 func (m *managerImpl) TargetShoot(ctx context.Context, shootName string) error {
 	return m.patchTarget(func(t *targetImpl) error {
+		if isDomainFormat(shootName) {
+			targetFlags, err := targetFlagsFromDomain(ctx, m, t, shootName)
+			if err != nil {
+				return err
+			}
+
+			if targetFlags != nil {
+				t.Garden = targetFlags.GardenName()
+				t.Project = targetFlags.ProjectName()
+				t.Shoot = targetFlags.ShootName()
+				return nil
+			}
+
+			return fmt.Errorf("failed to target shoot with url: %s", shootName)
```

Comment:
Nit: I'd prefer to turn it around

```suggestion
			if targetFlags == nil {
				return fmt.Errorf("failed to target shoot with url: %s", shootName)
			}

			t.Garden = targetFlags.GardenName()
			t.Project = targetFlags.ProjectName()
			t.Shoot = targetFlags.ShootName()
			
			return nil
```